File: mapView1/mapView1/ViewController.m
//
//  ViewController.m
//  mapView1
//
//  Created by Raunak Talwar on 8/20/15.
//  Copyright (c) 2015 Raunak Talwar. All rights reserved.
//

#import "ViewController.h"
#import <MapKit/MapKit.h>


@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)setMap:(id)sender {
    
    switch (((UISegmentedControl *)sender).selectedSegmentIndex) {
        case 0:
            self.mapView.mapType = MKMapTypeStandard;
            break;
        case 1:
            self.mapView.mapType = MKMapTypeSatellite;
            break;

        default:
            break;
    }
}
@end

File: Learn-Swift-1/Learn-Swift-1/ViewController.swift
//
//  ViewController.swift
//  Learn-Swift-1
//
//  Created by Raunak Talwar on 9/14/15.
//  Copyright © 2015 Raunak Talwar. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


    @IBOutlet weak var myLabel: UILabel!
    @IBAction func buttonPressed(sender: AnyObject) {
        
        let title = sender.titleForState(.Normal)!
        
        myLabel.text = "You clicked \(title) button"
        
        c();
    }
    
    var c: () -> () = {
        print("Closure is called")
    }
}

